File: simple_linear_regression.r
#Simple Linear Regression
#Importing the dataset
dataset = read.csv('Salary_Data.csv')


#Splitting data set into training set and test set
#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split== TRUE)
test_set = subset(dataset, split== FALSE)

#Feature Scaling
#training_set[, 2:3] = scale(training_set[, 2:3])
#test_set[, 2:3] = scale(test_set[, 2:3])

#Fitting simple linear regression model in R
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)

#Predicting the test set results
y_predict = predict(regressor, newdata = test_set)

#Visualizing the training set results
install.packages('ggplot2')
library(ggplot2)
ggplot() + geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
                      color = 'red') + 
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary vs Experience (Training Set)') + xlab('Years of Experience') +
  ylab('Salary')

#Visualizing the test set results
library(ggplot2)
ggplot() + geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
                      color = 'red') + 
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary vs Experience (Test Set)') + xlab('Years of Experience') +
  ylab('Salary')

File: simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Thu Jan  5 21:54:07 2017

@author: AmoolyaD
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#importing data sets
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
Y = dataset.iloc[:, 1].values

#Splitting the dataset into training set and test set
from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 1/3, random_state = 0)

#feature scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)￼
X_test = sc_X.transform(X_test)"""

#Fitting simple regression model to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, Y_train)

#predicting the test set results
y_pred = regressor.predict(X_test)

#Visualizing the training set results
plt.scatter(X_train, Y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary Vs Experience(Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

#Visualizing the test set results
plt.scatter(X_test, Y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary Vs Experience(Test Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()